src/main.ts

```diff
@@ -165,13 +205,45 @@ actionsToolkit.run(
 
     await core.group(`Booting builder`, async () => {
       const inspectCmd = await toolkit.buildx.getCommand(await context.getInspectArgs(inputs, toolkit));
-      await Exec.getExecOutput(inspectCmd.command, inspectCmd.args, {
-        ignoreReturnCode: true
-      }).then(res => {
-        if (res.stderr.length > 0 && res.exitCode != 0) {
-          throw new Error(res.stderr.match(/(.*)\s*$/)?.[0]?.trim() ?? 'unknown error');
-        }
-      });
+      
+      try {
+        await retryWithBackoff(
+          async () => {
+            // Create a promise that will timeout after 15 seconds
+            const timeoutPromise = new Promise<never>((_, reject) => {
+              setTimeout(() => {
+                reject(new Error('Timeout exceeded while waiting for buildkit to initialize'));
+              }, 15000); // 15 second timeout
+            });
```

Comment:
We don't need this timeout promise, `inspect` command already have some timeout: https://github.com/docker/buildx/blob/ea2b7020a4645bff395eb49e4e87ef08ba24eb93/commands/inspect.go#L38-L40